['Izzah Bte Abdul Jalil] iP

izzahaj opened the pull request:
# DukePro
> “Edit your life frequently and ruthlessly. It's your masterpiece after all” – Nathan W. Morris ([source](https://dansilvestre.com/productivity-quotes/))

DukePro is the 🔪 ruthlessly 🔪 quick paintbrush for the masterpiece of your life. It's:
- text-based
- easy to learn
- ~~FAST~~ UNBELIEVABLY FAST to use

All you need to do is:
1. download it from [here](https://nus-cs2103-ay2223s1.github.io/website/schedule/week4/project.html)
2. double-click it
3. add your tasks
4. let it manage your tasks for you

And it is FREE!

Features:
- [x] Managing tasks
- [ ] Managing deadlines \(coming soon)
- [ ] Reminders \(coming soon)
***
If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

izzahaj committed:
Implemented skeletal version

izzahaj committed:
Implemented skeletal version

izzahaj committed:
Added ability to store and list input text

izzahaj committed:
Added ability to store and list input text

izzahaj committed:
added ability to mark tasks as done

izzahaj committed:
added ability to mark tasks as done

izzahaj committed:
added todos, deadlines and events

izzahaj committed:
added todos, deadlines and events

izzahaj committed:
added i/o redirection technique for testing

izzahaj committed:
added i/o redirection technique for testing

izzahaj committed:
Added exceptions for missing description, time and unknown commands

izzahaj committed:
Added exceptions for missing description, time and unknown commands

izzahaj committed:
implemented deletion of tasks

izzahaj committed:
implemented deletion of tasks

izzahaj committed:
added deletion of tasks

izzahaj committed:
added deletion of tasks

izzahaj committed:
Added enums for a few exception messages

izzahaj committed:
Added enums for a few exception messages

izzahaj committed:
Separate setDone into mark and unmark methods

izzahaj committed:
Separate setDone into mark and unmark methods

izzahaj committed:
Add functionality to save and load created tasks

izzahaj committed:
Add functionality to save and load created tasks

izzahaj committed:
Refactor parsing of event and deadline commands

izzahaj committed:
Refactor parsing of event and deadline commands

izzahaj committed:
Refactor deadline and event classes to store dates instead of strings

izzahaj committed:
Change location of the saved task file

izzahaj committed:
Change location of the saved task file

izzahaj committed:
Revert "Change location of the saved task file"

This reverts commit a9613595f5d8b5910ba43943a721580262ce23a7.

izzahaj committed:
Revert "Change location of the saved task file"

This reverts commit a9613595f5d8b5910ba43943a721580262ce23a7.

izzahaj committed:
Change directory of saved file to project root

izzahaj committed:
Merge branch 'branch-Level-7'

izzahaj committed:
Resolve conflicts between master and branch-Level-8 branches

izzahaj committed:
Delete data directory

izzahaj committed:
Organise logic into command, parser, storage and ui classes

izzahaj committed:
Organise logic into command, parser, storage and ui classes

izzahaj committed:
Organise classes into packages

izzahaj committed:
Add gradle support

izzahaj committed:
Add gradle to project

izzahaj committed:
Add JUnit tests for parser and task classes

izzahaj committed:
Resolve conlficts from an earlier commit

izzahaj committed:
Create jar file

izzahaj committed:
Add JavaDoc comments

izzahaj committed:
Tweak to comply with coding standards

izzahaj committed:
Implement command to search for a task

izzahaj committed:
Merge branch 'branch-A-JavaDoc'

izzahaj committed:
Merge branch-A-CodingStandard to master

izzahaj committed:
Merge branch-Level-9 to master

izzahaj committed:
Add Checkstyle using Gradle

izzahaj committed:
Merge branch 'branch-A-CheckStyle'

izzahaj committed:
Add JavaFX dependencies

izzahaj committed:
Add GUI

izzahaj committed:
Merge branch 'branch-Level-10'

kangzongxian commented on src/main/java/mort/Mort.java:
```diff
@@ -0,0 +1,60 @@
+package mort;
```
I think overall everything looks really great and the Javadocs are there, nice!

kangzongxian commented on src/main/java/mort/command/DeleteCommand.java:
```diff
@@ -0,0 +1,32 @@
+package mort.command;
```
I find it quite good that you broke down the commands into different classes, making each file much more manageable!

kangzongxian commented on src/main/java/mort/parser/Parser.java:
```diff
@@ -0,0 +1,210 @@
+        String strippedCommand = fullCommand.stripTrailing();
+        String[] parsedCommand = strippedCommand.split(" ", 2);
+        try {
+            switch (parsedCommand[0]) {
```
Really good use of the switch statement here due to the numerous cases involved!

kangzongxian commented on src/main/java/mort/parser/Parser.java:
```diff
@@ -0,0 +1,210 @@
+    public static String[] parseSavedTask(String savedTask) {
+        return savedTask.split(" \\| ", 4);
+    }
+    private static Command validateToDo(String[] parsedCommand) throws MortException {
```
I think that Javadocs may be missing from this part here!

kangzongxian commented on src/main/java/mort/storage/Storage.java:
```diff
@@ -0,0 +1,110 @@
+     * @return List of tasks from the given text file.
+     * @throws MortException If task list cannot be loaded.
+     */
+    public ArrayList<Task> load() throws MortException {
```
I feel that it may be better to add some comments at different parts of this method here, since it is quite big, and adding comments may help with the understanding!

kangzongxian commented on src/main/java/mort/task/Task.java:
```diff
@@ -0,0 +1,98 @@
+/**
+ * Class that represents a task.
+ */
+public class Task {
```
Should this perhaps be an abstract class, since we might not be able to create a Task without knowing what type it is?

kangzongxian commented on src/main/java/mort/ui/Ui.java:
```diff
@@ -0,0 +1,103 @@
+        }
+    }
+    
+    public String getMissingTaskError(CommandWord commandWord, int num) {
```
I think maybe we should put JavaDocs here as well?

izzahaj committed:
Make Task class abstract

izzahaj committed:
Add assertions

izzahaj committed:
Storage load method: extract methods three helper methods createTask,
createDeadline and create Event

The Storage load method involves the creation of different types of tasks,
which makes the code very long and unreadable.

Extracting the creation of each type of task allows us to abstract away the
details according to SLAP and replace the long lines of code with multiple
short methods for better readability.

Let's
* move the code fragment that creates the Task to a new method called createTask
* further split the createTask method into two methods: createEvent and createDeadline
* createEvent contains the code fragment for creating events
* createDeadline contains the code fragment for creating deadlines

izzahaj committed:
Use streams to filter tasks containing keyword in find method of TaskList class

izzahaj committed:
Merge pull request #1 from izzahaj/branch-A-Assertions

A-Assertions

izzahaj committed:
Merge master to branch-A-CodeQuality

izzahaj committed:
Merge pull request #2 from izzahaj/branch-A-CodeQuality

A-CodeQuality

izzahaj committed:
Merge branch 'master' into branch-A-Streams

izzahaj committed:
Merge pull request #3 from izzahaj/branch-A-Streams

A-Streams

izzahaj committed:
Add continuous integration

izzahaj committed:
Implement command to view tasks in the form of a schedule

This feature allows users with many deadlines or events due or
scheduled on a given date.

izzahaj committed:
Merge pull request #5 from izzahaj/branch-B-ViewSchedules

B-ViewSchedules

izzahaj committed:
Add JavaDocs to DeleteCommand and MainWindow classes

izzahaj committed:
Change order of imports in test classes.

izzahaj committed:
Add JUnit tests for task classes and Ui class

izzahaj committed:
Add credit for reused work

izzahaj committed:
Decrease padding of dialog box

izzahaj committed:
Add a user guide

izzahaj committed:
Add product screenshot

izzahaj committed:
Remove horizontal lines in user guide

izzahaj committed:
Reorder table of contents in user guide

izzahaj committed:
Fix JavaFX version bugs on Windows and Linux

izzahaj committed:
Remove unused JavaFX Font import

izzahaj committed:
Add resizing of window

izzahaj committed:
Update credits for reused code
